test/v1/controllers/validators/AmendOtherReliefsValidatorFactorySpec.scala

```diff
@@ -0,0 +1,502 @@
+    postCessationTradeReliefAndCertainOtherLosses = Some(Seq(parsedPostCessationTradeReliefAndCertainOtherLosses)),
+    annualPaymentsMade = Some(parsedAnnualPaymentsMade),
+    qualifyingLoanInterestPayments = Some(Seq(parsedQualifyingLoanInterestPayments))
+  )
+
+  private val validatorFactory = new AmendOtherReliefsValidatorFactory(mockAppConfig)
+
+  private def validator(nino: String, taxYear: String, body: JsValue) = validatorFactory.validator(nino, taxYear, body)
+
+  "running a validation" should {
+    "return no errors" when {
+      "a valid request is supplied" in {
+        val result: Either[ErrorWrapper, AmendOtherReliefsRequestData] =
+          validator(validNino, validTaxYear, validBody).validateAndWrapResult()
+
+        result shouldBe Right(AmendOtherReliefsRequestData(parsedNino, parsedTaxYear, parsedBody))
+      }
+    }
+    "return NinoFormatError" when {
+      "an invalid nino is supplied" in {
+        val result: Either[ErrorWrapper, AmendOtherReliefsRequestData] =
+          validator("invalid", validTaxYear, validBody).validateAndWrapResult()
+
+        result shouldBe Left(ErrorWrapper(correlationId, NinoFormatError))
+      }
+    }
+    "return TaxYearFormatError" when {
+      "an invalid tax year is supplied" in {
+        val result: Either[ErrorWrapper, AmendOtherReliefsRequestData] =
+          validator(validNino, "201831", validBody).validateAndWrapResult()
+
+        result shouldBe Left(ErrorWrapper(correlationId, TaxYearFormatError))
+      }
+    }
+    "return RuleTaxYearRangeInvalidError" when {
+      "the tax year range exceeds 1" in {
+        val result: Either[ErrorWrapper, AmendOtherReliefsRequestData] =
+          validator(validNino, "2021-24", validBody).validateAndWrapResult()
+
+        result shouldBe Left(ErrorWrapper(correlationId, RuleTaxYearRangeInvalidError))
+      }
+    }
+    "return RuleTaxYearNotSupportedError" when {
+      "the given tax year is before the minimum tax year" in {
+        val result: Either[ErrorWrapper, AmendOtherReliefsRequestData] =
+          validator(validNino, "2019-20", validBody).validateAndWrapResult()
+
+        result shouldBe Left(ErrorWrapper(correlationId, RuleTaxYearNotSupportedError))
+      }
+    }
+    "return multiple errors" when {
+      "request supplied has multiple errors" in {
+        val result: Either[ErrorWrapper, AmendOtherReliefsRequestData] =
+          validator("invalid", "invalid", validBody).validateAndWrapResult()
+
+        result shouldBe Left(
+          ErrorWrapper(
+            correlationId,
+            BadRequestError,
+            Some(List(NinoFormatError, TaxYearFormatError))
+          )
+        )
+      }
+    }
+    "return a RULE_INCORRECT_OR_EMPTY_BODY_SUBMITTED error" when {
+      "an empty JSON body is submitted" in {
+        val result: Either[ErrorWrapper, AmendOtherReliefsRequestData] =
+          validator(validNino, validTaxYear, emptyJson).validateAndWrapResult()
```

Comment:
Don't need to create and use the emptyJson. Can instead just use `JsObject.empty`